### ISSUE TITLE
Some errors highlight the line past the error, not the line with the error.

### ISSUE THREAD

**Issue by RefinedSoftwareLLC** [godotengine/godot#1197]
Forget to type a ":" and look at the error message:

error(14,0): ':' expected at end of line.

Except that it is not expected at the end of line 14, it is expected at the end of line 13. This error needs to highlight the previous line then it currently does. I am teaching others how to use goDotEngine and the highlighted line having nothing to do with the error is a constant source of frustration.

**Comment by reduz** [comment]
not neccesarily, i fyou forget : at the end of say, an if (), the parser
will assume the expression continues on the next line (similar to python),
so in practice it _is_ expected in the next line

**Comment by RefinedSoftwareLLC** [comment]
Then highlight them both!

**Comment by reduz** [comment]
makes sense, since one is able to tell an expression started in one line
and ended on other.
or actually i think for the specific error of missing : , it makes sense
that it shows the first line, you are right

**Comment by RefinedSoftwareLLC** [comment]
Multi-line conditions have to be wrapped between "(" and ")" right? so if there is no "(" then it should not assume the next line has anything to do with it.

**Comment by RefinedSoftwareLLC** [comment]
The error message should probably also say that multi-line if/thens must be wrapped in "()" for those not knowing why it isn't working.

**Comment by RefinedSoftwareLLC** [comment]
100 if true == false
101 >>    and true == false:


currently this shows:
error(101:0): ':' expected at end of line.

But it should show this instead:
error(100:0): ':' expected at end of line 100 or multi-line Boolean expression should be wrapped in "()"

**Comment by akien-mga** [comment]
I confirm the issue and it's true that it can be misleading, especially since adding ":" on the designated line does not solve the issue, nor trigger another one, e.g. this code will still produce the same error:
```100 if (true)
101     print("oops")::::::::
```
```
error(101:0): ':' expected at end of line.
```

**Comment by akien-mga** [comment]
Fixed by #2955.
